core/src/com/unciv/ui/worldscreen/unit/UnitActionsTable.kt

```diff
@@ -41,6 +41,7 @@ class UnitActionsTable(val worldScreen: WorldScreen) : Table() {
                 "Move unit" -> return UnitIconAndKey(ImageGetter.getStatIcon("Movement"))
                 "Stop movement" -> return UnitIconAndKey(ImageGetter.getStatIcon("Movement").apply { color = Color.RED }, '.')
                 "Promote" -> return UnitIconAndKey(ImageGetter.getImage("OtherIcons/Star").apply { color = Color.GOLD }, 'o')
+                "Upgrade" -> return UnitIconAndKey(ImageGetter.getImage("OtherIcons/Star").apply { color = Color.GOLD })
```

Comment:
You will never have an Upgrade action, always an Upgrade to, which is caught on line 26